src/confluent_kafka/aio/_AIOProducer.py

```diff
@@ -60,47 +142,254 @@ async def _call(self, blocking_task, *args, **kwargs):
+        - Ensures immediate callback processing
+        - Lower latency for delivery confirmation
+        """
+        # Call produce_batch first
+        self._producer.produce_batch(target_topic, batch_messages, on_delivery=batch_delivery_callback)
+        
+        # Immediately poll to process delivery callbacks in the same worker
+        poll_result = self._producer.poll(0)
+        
+        return poll_result
+
+    async def _flush_buffer(self, target_topic=None):
+        """Flush the current message buffer using produce_batch via thread pool
+        
+        This method now properly handles messages for different topics by grouping
+        them and calling produce_batch separately for each topic.
+        """
+        if not self._message_buffer:
+            return
+        
+        # Group messages by topic
+        topic_groups = {}
+        for i, msg_data in enumerate(self._message_buffer):
+            topic = msg_data['topic']
+            if topic not in topic_groups:
+                topic_groups[topic] = {
+                    'messages': [],
+                    'futures': [],
+                    'callbacks': [],
+                    'indices': []
+                }
+            topic_groups[topic]['messages'].append(msg_data)
+            topic_groups[topic]['futures'].append(self._buffer_futures[i])
+            topic_groups[topic]['callbacks'].append(msg_data.get('user_callback'))
+            topic_groups[topic]['indices'].append(i)
+        
+        # Clear buffers immediately (reduce memory pressure)
+        self._message_buffer.clear()
+        self._buffer_futures.clear()
+        
+        # Process each topic group separately
+        for topic, group_data in topic_groups.items():
+            # Skip if target_topic is specified and doesn't match
+            if target_topic is not None and topic != target_topic:
+                # Re-add messages for other topics back to buffer
+                for j, msg_data in enumerate(group_data['messages']):
+                    self._message_buffer.append(msg_data)
+                    self._buffer_futures.append(group_data['futures'][j])
+                continue
+            
+            # Convert messages to batch format
+            batch_messages = []
+            for msg_data in group_data['messages']:
+                batch_msg = {
+                    'value': msg_data['value'],
+                    'key': msg_data['key']
+                }
+                
+                # Add optional fields if present
+                if 'partition' in msg_data:
+                    batch_msg['partition'] = msg_data['partition']
+                if 'timestamp' in msg_data:
+                    batch_msg['timestamp'] = msg_data['timestamp']
+                if 'headers' in msg_data:
+                    batch_msg['headers'] = msg_data['headers']
+                
+                batch_messages.append(batch_msg)
```

Comment:
Why can't we just append the original `msg_data` or pass `group_data['messages']` directly? If we really need to avoid reference mutation why not just `copy.copy` or `copy.deepcopy`?